ui/src/components/layout/Markdown.vue

```diff
@@ -1,190 +1,1018 @@
     import * as Markdown from "../../utils/markdown";
 
-    const props = withDefaults(defineProps<{
-        source?: string;
-        permalink?: boolean;
-        fontSizeVar?: string;
-        html?: boolean;
-    }>(), {
-        source: "",
-        permalink: false,
-        fontSizeVar: "font-size-sm",
-        html: true
-    });
-
-    const markdownRenderer = ref()
-
-    const sourceWithReplacedAlerts = computed(() => {
-        return props.source.replace(
+    /**
+     * Unified component props
+     */
+    const props = withDefaults(
+        defineProps<{
+            source?: string;
+            permalink?: boolean;
+            fontSizeVar?: string;
+            html?: boolean;
+            showSearch?: boolean;
+            collapseExamples?: boolean;
+            variant?: "default" | "enhanced";
+        }>(),
+        {
+            source: "",
+            permalink: false,
+            fontSizeVar: "font-size-sm",
+            html: true,
+            showSearch: false, // good default for OSS docs
+            collapseExamples: false,
+            variant: "enhanced",
+        }
+    );
+
+    /**
+     * Reactive state
+     */
+    const markdownHtml = ref<string>("");
+    const searchQuery = ref("");
+    const markdownContainer = ref<HTMLElement | null>(null);
+    const cleanups: Array<() => void> = [];
+    const hasMatchesRef = ref(true);
+
+    /**
+     * Derived values
+     */
+    const fontSizeCss = computed(() => `var(--${props.fontSizeVar})`);
+    const hasMatches = computed(() => hasMatchesRef.value);
+
+    /**
+     * Replace custom alert syntax (legacy component behavior)
+     * ::alert{type="info"} ... ::: -> ::: info ... :::
+     */
+    function replaceAlertSyntax(source: string | undefined) {
+        if (!source) return "";
+        return source.replace(
             /(\n)?:\s*:\s*alert\{type="(.*?)"\}\s*\n([\s\S]*?)\n:\s*:(\n)?/g,
-            (_, newLine1, type, content, newLine2) => `${newLine1 ?? ""}::: ${type}\n${content}\n:::${newLine2 ?? ""}`
+            (_, newLine1, type, content, newLine2) =>
+                `${newLine1 ?? ""}::: ${type}\n${content}\n:::${newLine2 ?? ""}`
         );
-    })
+    }
 
+    /**
+     * RENDERING
+     */
+    watch(
+        () => props.source,
+        async () => {
+            await renderMarkdown();
+        },
+        {immediate: true}
+    );
+
+    watch(searchQuery, () => {
+        applySearchFilter();
+    });
+
+    onBeforeUnmount(() => {
+        cleanupEnhancements();
+    });
+
+    /**
+     * Main render function
+     */
     async function renderMarkdown() {
-        return await Markdown.render(sourceWithReplacedAlerts.value, {
+        if (typeof window === "undefined") {
+            markdownHtml.value = props.source ?? "";
+            return;
+        }
+
+        const source = replaceAlertSyntax(props.source);
+
+        markdownHtml.value = await Markdown.render(source, {
```

Comment:
```suggestion
        markdownHtml.value = await Markdown.render(props.source, {
```

This line should not need to replace anything since all the replacement is now done in the function